modules/loader.js

```diff
@@ -559,6 +567,131 @@ __private.loadBlockChain = function() {
 		});
 };
 
+/**
+ * Validate given block
+ *
+ * @param {object} block
+ * @param {function} cb
+ * @returns {setImmediateCallback} cb, err
+ */
+__private.validateBlock = (blockToVerify, cb) => {
+	const lastBlock = modules.blocks.lastBlock.get();
+
+	modules.blocks.utils.loadBlockByHeight(
+		blockToVerify.height - 1,
+		(lastBlockError, secondLastBlockToVerify) => {
+			if (lastBlockError) {
+				return setImmediate(cb, lastBlockError);
+			}
+
+			// Set the block temporarily for block verification
+			modules.blocks.lastBlock.set(secondLastBlockToVerify);
```

Comment:
Does `verifyBlock` not work if last block is not set here?

Reply:
Yep, it will fail.

Reply:
That is bad. It should be fixed.

Reply:
`verifyBlock` function is developed with logic to verify only blocks when last block is one step behind that block. Yes it should be updated but out of scope for this PR.